lib/reporter/results.js

```diff
@@ -2,6 +2,11 @@ const AssertionError = require('assertion-error');
 const Utils = require('../utils');
 const {Logger} = Utils;
 
+const allowedHooks = {
+  'after_hook': true,
```

Comment:
I'm not sure why we need this here?

Reply:
I need to figure this out, we can't break the current report structure hence the implementation will have to change.

Reply:
reworked and fixed.